packages/gguf/src/types.ts

```diff
@@ -75,21 +86,43 @@ interface Tokenizer {
 	"tokenizer.ggml.bos_token_id": number;
 	"tokenizer.ggml.eos_token_id": number;
 	"tokenizer.ggml.add_bos_token": boolean;
-	"tokenizer.chat_template": string;
+	"tokenizer.chat_template"?: string;
 }
+interface NoTokenizer {
+	"tokenizer.ggml.model"?: undefined;
+}
+
+/// Models outside of llama.cpp: "rwkv" and "whisper"
 
-export type RWKV = ModelBase<"rwkv"> & { "rwkv.architecture_version": number };
-export type LLM = TransformerLLM | RWKV;
-export type Whisper = ModelBase<"encoder.whisper"> & ModelBase<"decoder.whisper">;
-export type Model = (LLM | Whisper) & Partial<Tokenizer>;
+export type RWKV = GGUFGeneralInfo<"rwkv"> &
+	ModelBase<"rwkv"> & {
+		"rwkv.architecture_version": number;
+	};
 
-export type GGUFMetadata = {
+// TODO: whisper.cpp doesn't yet support gguf. This maybe changed in the future.
+export type Whisper = GGUFGeneralInfo<"whisper"> &
+	ModelBase<"encoder.whisper"> &
+	ModelBase<"decoder.whisper"> & {
+		"whisper.encoder.mels_count": number;
+		"whisper.encoder.attention.head_count": number;
+		"whisper.decoder.attention.head_count": number;
+	};
+
+/// Types for parse output
+
+export enum GGUFType {
+	STRICT,
+	NON_STRICT,
+}
+
+export type GGUFMetadata<TGGUFType extends GGUFType = GGUFType.STRICT> = {
```

Comment:
We should probably switch to something like https://github.com/sindresorhus/type-fest/blob/main/source/except.d.ts

```ts
interface GGUFMetadataOptions {
	/**
   * ...
   * 
	 * @default true
   */
  strict: boolean;
}


export GGUFMetadata<Options extends GGUFMetadataOptions = { strict: true}> {
  ...
```

Reply:
Good idea, I've never thought about this before. Implemented in https://github.com/huggingface/huggingface.js/pull/655/commits/31bac8bbba365e13c517c48452fdee1d5843abd3